File: 383-ransom-note/ransom-note.py
class Solution(object):
    def canConstruct(self, ransomNote, magazine):
        """
        :type ransomNote: str
        :type magazine: str
        :rtype: bool
        """
        magazin_hashmap=defaultdict(int)
        for i in range(len(magazine)):
            magazin_hashmap[magazine[i]]+=1
        print(magazin_hashmap)
        for i in range(len(ransomNote)):
            if ransomNote[i] in magazin_hashmap and magazin_hashmap[ransomNote[i]]>0:
                magazin_hashmap[ransomNote[i]]-=1
            else:
                return False
        return True
        print(magazin_hashmap)
        

        print(magazin_hashmap)
        

File: 383-ransom-note/ransom-note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        HashMap<Character,Integer> magazinMap = new HashMap<>();
        for(int i=0;i<magazine.length();i++){
            char item=magazine.charAt(i);
            magazinMap.put(item,magazinMap.getOrDefault(item,0)+1);
        }
        for(int i=0;i<ransomNote.length();i++){
            char item=ransomNote.charAt(i);
            if(magazinMap.containsKey(item) && magazinMap.get(item)>0){
                magazinMap.put(item,magazinMap.getOrDefault(item,0)-1);
            }else{
                return false;
            }
        }
        return true;
    }
}